R/conos.R

```diff
@@ -743,6 +782,71 @@ stableTreeClusters <- function(refwt,tests,min.threshold=0.8,min.size=10,n.cores
   }
 }
 
+#' Create and preprocess a Seurat object
+#'
+#' @description Create Seurat object from gene count matrix
+#'
+#' @param count.matrix gene count matrix
+#' @param vars.to.regress variables to regress with Seurat
+#' @param verbose verbose mode
+#' @param do.par use parallel processing for regressing out variables faster, for
+#' Seurat v3, use \code{\link[future]{plan}} instead
+#' @param n.pcs number of principal components
+#' @param cluster do clustering
+#' @param tsne do tSNE embedding, for Seurat v3, will default to UMAP if available
+#'
+#' @return Seurat object
+#'
+#'
+#' @export
+#'
+basicSeuratProc <- function(count.matrix, vars.to.regress=NULL, verbose=TRUE, do.par=TRUE, n.pcs=100, cluster=TRUE, tsne=TRUE) {
+  if (!requireNamespace("Seurat", quietly = TRUE)) {
+    stop("You need to install 'Seurat' package to be able to use this function")
+  }
+  if (packageVersion('Seurat') < package_version(x = '3.0.0')) {
+    if (verbose) {
+      warning("Running Seurat v2 workflow", call. = FALSE, immediate. = TRUE)
+    }
+    rownames(count.matrix) <- make.unique(rownames(count.matrix))
+    max.n.pcs <- min(nrow(count.matrix) - 1, ncol(count.matrix) - 1, n.pcs)
+    so <- Seurat::CreateSeuratObject(count.matrix, display.progress=verbose) %>%
+      Seurat::NormalizeData(display.progress=verbose) %>%
+      Seurat::ScaleData(vars.to.regress=vars.to.regress, display.progress=verbose, do.par=do.par) %>%
+      Seurat::FindVariableGenes(do.plot = FALSE, display.progress=verbose) %>%
+      Seurat::RunPCA(pcs.compute=max.n.pcs, do.print=FALSE)
+    if (cluster) {
+      so <- Seurat::FindClusters(so, n.iter=500, n.start=10, dims.use=1:n.pcs, print.output = F)
+    }
+    if (tsne) {
+      so <- Seurat::RunTSNE(so, dims.use=1:n.pcs)
+    }
+  } else {
+    if (verbose) {
+      message("Running Seurat v3 workflow")
+    }
+    so <- Seurat::CreateSeuratObject(counts = count.matrix)
+    so <- Seurat::NormalizeData(object = so, verbose = verbose)
+    so <- Seurat::FindVariableFeatures(object = so, verbose = verbose)
+    so <- Seurat::ScaleData(object = so, vars.to.regress = vars.to.regress, verbose = verbose)
+    so <- Seurat::RunPCA(object = so, npcs = n.pcs, verbose = verbose)
+    if (cluster) {
+      so <- Seurat::FindNeighbors(object = so, dims = 1:n.pcs, verbose = verbose)
+      so <- Seurat::FindClusters(object = so, n.iter = 500, n.start = 10, verbose = verbose)
+    }
+    if (tsne) {
+      so <- tryCatch(
```

Comment:
Don't you think that calling UMAP when `tsne=T` is confusing? Moreover, don't think that everyone prefer this algorithm. So, I think, if one want to run UMAP, they can call `RunUMAP` by hands. Or, if you think it's an often case, you can add option `umap=F` to the function.

Reply:
Agreed. Changed with a new `umap` parameter, also enabled use of `umap` for Seurat v2.3.1, when `RunUMAP was introduced, and higher.

Reply:
Cool! One more comment about UMAP: you set it by default in `plotSamples` (plot.R) instead of t-SNE. Could you please make behavior consistent there, as well?